src/vs/platform/environment/node/argv.ts

```diff
@@ -217,6 +219,38 @@ export function buildVersionMessage(version: string | undefined, commit: string
 	return `${version || localize('unknownVersion', "Unknown version")}\n${commit || localize('unknownCommit', "Unknown commit")}\n${process.arch}`;
 }
 
+export function buildTelemetryMessage(appRoot: string, extensionsPath: string): string {
+	try {
+		// Gets all the directories inside the extension directory
+		const dirs = readdirSync(extensionsPath).filter(files => statSync(join(extensionsPath, files)).isDirectory());
+		const telemetryJsonFolders: string[] = [];
+		dirs.forEach((dir) => {
+			const files = readdirSync(join(extensionsPath, dir)).filter(file => file === 'telemetry.json');
+			// We know it contains a telemetry.json file so we add it to the list of folders which have one
+			if (files.length === 1) {
+				telemetryJsonFolders.push(dir);
+			}
+		});
+		const mergedTelemetry = Object.create(null);
+		// Simple function to merge the telemetry into one json object
+		const mergeTelemetry = (contents: string, dirName: string) => {
+			const telemetryData = JSON.parse(contents);
+			mergedTelemetry[dirName] = telemetryData;
+		};
+		telemetryJsonFolders.forEach((folder) => {
+			const contents = readFileSync(join(extensionsPath, folder, 'telemetry.json')).toString();
+			mergeTelemetry(contents, folder);
+		});
+		let contents = readFileSync(join(appRoot, 'out/', 'telemetry-core.json')).toString();
+		mergeTelemetry(contents, 'vscode-core');
+		contents = readFileSync(join(appRoot, 'out/', 'telemetry-extensions.json')).toString();
+		mergeTelemetry(contents, 'vscode-extensions');
+		return JSON.stringify(mergedTelemetry, null, 4);
+	} catch (err) {
+		return 'Unable to read VS Code telemetry events!';
```

Comment:
Just remove the whole `try catch` block and let the error bubble up and be printed to the console.